solr/solr-ref-guide/src/rate-limiters.adoc

```diff
@@ -0,0 +1,116 @@
+pronounced under high stress in production workloads.
+
+== Rate Limiter Configurations
+The default rate limiter is search rate limiter. Accordingly, it can be configured in web.xml under initParams for
+SolrRequestFilter.
+
+[source,xml]
+----
+<filter-name>SolrRequestFilter</filter-name>
+----
+
+==== Enable Query Rate Limiter
+Controls enabling of query rate limiter. Default value is false.
+[source,xml]
+----
+<param-name>isQueryRateLimiterEnabled</param-name>
+----
+[source,xml]
+----
+<param-value>true</param-value>
+----
+
+==== Maximum Number Of Concurrent Requests
+Allows setting maximum concurrent search requests at a given point in time. Default value is number of cores * 3.
+[source,xml]
+----
+<param-name>maxQueryRequests</param-name>
+----
+[source,xml]
+----
+<param-value>15</param-value>
+----
+
+==== Request Slot Allocation Wait Time
+Wait time in ms for which a request will wait for a slot to be available when all slots are full,
+before the request is put into the wait queue. This allows requests to have a chance to proceed if
+the unavailability of the request slots for this rate limiter is a transient phenomenon. Default value
+is -1, indicating no wait. 0 will represent the same -- no wait.
+[source,xml]
+----
+<param-name>queryWaitForSlotAllocationInMS</param-name>
+----
+[source,xml]
+----
+<param-value>100</param-value>
+----
+
+==== Slot Borrowing Enabled
+If slot borrowing (described below) is enabled or not. Default value is false.
+[source,xml]
+----
+<param-name>queryAllowSlotBorrowing</param-name>
+----
+[source,xml]
+----
+<param-value>true</param-value>
+----
+
+==== Guaranteed Slots
+The number of guaranteed slots that the query rate limiter will reserve irrespective
+of the load of query requests. This is used only if slot borrowing is enabled and acts
+as the threshold beyond which query rate limiter will not allow other request types to
+borrow slots from its quota. Default value is allowed number of concurrent requests / 2.
+[source,xml]
+----
+<param-name>queryGuaranteedSlots</param-name>
+----
+[source,xml]
+----
+<param-value>200</param-value>
+----
+
+== Salient Points
+
+These are some of the things to keep in mind when using rate limiters
+
+=== Over Subscribing
+It is possible to define a size of quota for a request type which exceeds the size
+of the available threadpool. Solr does not enforce rules on the size of a quota that
+can be define for a request type. This is intentionally done to allow users full
+control on their quota allocation. However, if the quota exceeds the available threadpool's
+size, the standard queuing policies of the threadpool will kick in.
+
+=== Slot Borrowing
+If a quota does not have backlog but other quotas do, then the relatively less busier quota can
+"borrow" slot from the busier quotas. This is done on a round robin basis today with a futuristic
```

Comment:
Is there a JIRA for this? Would be good to link to it if it exists (or create it if it does not)